CI is broken on loading / using frozen images on x86

### Description

relates to:

- https://github.com/moby/moby/pull/50292#issuecomment-3161847227
- https://github.com/moby/moby/pull/50645#issuecomment-3161726006


CI appears to be broken w.r.t. the frozen images; could be either some path issue, or something else (could newer Dockerfile front-end use the new fancy sub-variants for x86 for example? `TARGETVARIANT`); https://github.com/moby/moby/blob/28df7815149571b3d3d32e26c32462a03b5b9d17/Dockerfile#L110-L120


Docker info doesn't show those;

```
   Kernel Version: 6.11.0-1018-azure
   Operating System: Ubuntu 24.04.2 LTS
   OSType: linux
   Architecture: x86_64
```

Integration tests;

```
Building test suite binary /go/src/github.com/docker/docker/integration/service/test.main
Building test suite binary /go/src/github.com/docker/docker/integration/session/test.main
Building test suite binary /go/src/github.com/docker/docker/integration/system/test.main
Building test suite binary /go/src/github.com/docker/docker/integration/volume/test.main
---> Making bundle: .integration-daemon-start (in bundles/test-integration)
Using test binary /usr/local/cli-integration/docker
# DOCKER_EXPERIMENTAL is set: starting daemon with experimental features enabled! 
Starting apparmor (via systemctl): apparmor.service.
Starting dockerd
INFO: Waiting for daemon to start...

Running integration-test (iteration 1)
Running /go/src/github.com/docker/docker/integration/build (amd64.docker.docker.integration.build) flags=-test.v -test.timeout=10m   -test.coverprofile=/go/src/github.com/docker/docker/bundles/test-integration/amd64-docker-docker-integration-build-coverage.out
panic: error loading frozen images: unable to resolve platform: unexpected end of JSON input

goroutine 1 [running, locked to thread]:
github.com/moby/moby/v2/integration/build.TestMain(0xc0000ce5a0)
	/go/src/github.com/docker/docker/integration/build/main_test.go:38 +0x1e9
main.main()
	_testmain.go:97 +0xa8

goroutine 12 [runnable]:
runtime.goexit1()
	/usr/local/go/src/runtime/proc.go:4313 +0x9d
runtime.goexit({})
	/usr/local/go/src/runtime/asm_amd64.s:1701 +0x6
created by go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp.(*client).contextWithStop in goroutine 7
	/go/src/github.com/docker/docker/vendor/go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp/client.go:382 +0xe5

goroutine 15 [IO wait]:
internal/poll.runtime_pollWait(0x7f895dfe8ed0, 0x72)
	/usr/local/go/src/runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0xc00022d680?, 0xc00051d000?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0xc00022d680, {0xc00051d000, 0x1000, 0x1000})
	/usr/local/go/src/internal/poll/fd_unix.go:165 +0x27a
net.(*netFD).Read(0xc00022d680, {0xc00051d000?, 0x408494?, 0x0?})
	/usr/local/go/src/net/fd_posix.go:55 +0x25
net.(*conn).Read(0xc00019a7c8, {0xc00051d000?, 0x40828f?, 0xc0005536e0?})
	/usr/local/go/src/net/net.go:194 +0x45
net/http.(*persistConn).Read(0xc0005505a0, {0xc00051d000?, 0x7f49c5?, 0x15c7500?})
	/usr/local/go/src/net/http/transport.go:2122 +0x47
bufio.(*Reader).fill(0xc00055c180)
	/usr/local/go/src/bufio/bufio.go:113 +0x103
bufio.(*Reader).Peek(0xc00055c180, 0x1)
	/usr/local/go/src/bufio/bufio.go:152 +0x53
net/http.(*persistConn).readLoop(0xc0005505a0)
	/usr/local/go/src/net/http/transport.go:2275 +0x172
created by net/http.(*Transport).dialConn in goroutine 13
	/usr/local/go/src/net/http/transport.go:1944 +0x174f

goroutine 16 [select]:
net/http.(*persistConn).writeLoop(0xc0005505a0)
	/usr/local/go/src/net/http/transport.go:2590 +0xe7
created by net/http.(*Transport).dialConn in goroutine 13
	/usr/local/go/src/net/http/transport.go:1945 +0x17a5

=== Failed
```

Tests;

```
=== RUN   TestBuildUserNamespaceValidateCapabilitiesAreV2
Loaded image: busybox:latest
Loaded image: busybox:glibc
Loaded image: debian:bookworm-slim
Error unpacking image debian:bookworm-slim: apply layer error for "docker.io/library/debian:bookworm-slim": wrong diff id calculated on extraction "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
Loaded image: hello-world:latest
Loaded image: arm32v7/hello-world:latest
Loaded image: hello-world:amd64
Loaded image: hello-world:arm64
    build_userns_linux_test.go:77: assertion failed: error is not nil: apply layer error for "": wrong diff id calculated on extraction "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
--- FAIL: TestBuildUserNamespaceValidateCapabilitiesAreV2 (0.89s)
FAIL
```

Error looks to come from containerd vendor; not sure why the `i.Name()` is empty, but maybe that's expected https://github.com/moby/moby/blob/0aa8fe0bf949cbccab74414cb091728854be0d7d/vendor/github.com/containerd/containerd/v2/client/image.go#L335-L338

Also see https://github.com/moby/moby/pull/50292#issuecomment-3161847227

```
panic: error loading frozen images: invalid diffID for layer 0:
expected "sha256:9c27e219663c25e0f28493790cc0b88bc973ba3b1686355f221c38a36978ac63",
got "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
```

So; `e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855` is the digest for `nil`; https://go.dev/play/p/FA77z66YhZh

```go
package main

import (
	"crypto/sha256"
	"fmt"
)

func main() {
	fmt.Printf("sha256:%x\n", sha256.Sum256(nil))
}
```

--- a/contrib/download-frozen-image-v2.sh
+++ b/contrib/download-frozen-image-v2.sh
@@ -81,7 +81,7 @@ fetch_blob() {
 			-D-
 	)"
 	curlHeaders="$(echo "$curlHeaders" | tr -d '\r')"
-	if grep -qE "^HTTP/[0-9].[0-9] 3" <<< "$curlHeaders"; then
+	if grep -qE "^HTTP/[0-9](.[0-9])* 3" <<< "$curlHeaders"; then
 		rm -f "$targetFile"
 
 		local blobRedirect

--- a/contrib/download-frozen-image-v2.sh
+++ b/contrib/download-frozen-image-v2.sh
@@ -72,30 +72,11 @@ fetch_blob() {
 	shift
 	local curlArgs=("$@")
 
-	local curlHeaders
-	curlHeaders="$(
-		curl -S "${curlArgs[@]}" \
-			-H "Authorization: Bearer $token" \
-			"$registryBase/v2/$image/blobs/$digest" \
-			-o "$targetFile" \
-			-D-
-	)"
-	curlHeaders="$(echo "$curlHeaders" | tr -d '\r')"
-	if grep -qE "^HTTP/[0-9](.[0-9])* 3" <<< "$curlHeaders"; then
-		rm -f "$targetFile"
-
-		local blobRedirect
-		blobRedirect="$(echo "$curlHeaders" | awk -F ': ' 'tolower($1) == "location" { print $2; exit }')"
-		if [ -z "$blobRedirect" ]; then
-			echo >&2 "error: failed fetching '$image' blob '$digest'"
-			echo "$curlHeaders" | head -1 >&2
-			return 1
-		fi
-
-		curl -fSL "${curlArgs[@]}" \
-			"$blobRedirect" \
-			-o "$targetFile"
-	fi
+	curl -L -S "${curlArgs[@]}" \
+		-H "Authorization: Bearer $token" \
+		"$registryBase/v2/$image/blobs/$digest" \
+		-o "$targetFile" \
+		-D-
 }
 
 # handle 'application/vnd.docker.distribution.manifest.v2+json' manifest

